enclave/core/sgx/calls.c

```diff
@@ -135,12 +136,65 @@ extern bool oe_disable_debug_malloc_check;
 **==============================================================================
 */
 
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+extern const oe_enclave_config_t* oe_enclave_config;
+
+static oe_result_t _load_config(const oe_enclave_config_t* config_from_host)
+{
+    const oe_eeid_page_t* eeid_page = __oe_get_eeid_page();
+    const oe_eeid_t* eeid = &eeid_page->eeid;
+
+    if (!eeid || eeid->version == 0)
+        return OE_OK;
+
+    /* With EEID, we expect a config */
+    if (!config_from_host)
+        return OE_INVALID_PARAMETER;
+
+    oe_enclave_config_t config = *config_from_host;
+
+    if (!config.data || !config.size)
+        return OE_OK; /* no or zero-size configs are ok */
+
+    /* Check that the config hashes to the correct value */
+    OE_SHA256 config_hash;
+    oe_sha256(config.data, config.size, &config_hash);
+    if (memcmp(config_hash.buf, eeid_page->config_id.buf, OE_SHA256_SIZE) != 0)
+        return OE_INVALID_PARAMETER;
```

Comment:
Consider a more descriptive error like `OE_VERIFY_FAILED`

Reply:
Yeah, I wasn't sure which one to pick, there isn't any that perfectly describes the error. (Changed to `OE_VERIFY_FAILED` now.)